how-to/workspace-platform-starter/client/src/framework/platform/interopbroker.ts

```diff
@@ -929,49 +941,63 @@ export function interopOverride(
 			// options the window can then use raiseIntent against a specific app (the selected one). this logic runs in
 			// the provider so we are using it as a way of determining the root (so it works with root hosting and
 			// subdirectory based hosting if a url is not provided)
-			const winOption = {
-				name: "intent-picker",
-				includeInSnapshot: false,
-				fdc3InteropApi: this._intentResolverOptions?.fdc3InteropApi,
-				defaultWidth: this._intentResolverOptions?.width,
-				defaultHeight: this._intentResolverOptions?.height,
-				showTaskbarIcon: false,
-				saveWindowState: false,
-				defaultCentered: true,
-				customData: {
-					title: this._intentResolverOptions?.title,
-					apps: launchOptions.apps,
-					intent: launchOptions.intent,
-					intents: launchOptions.intents,
-					unregisteredAppId: this._unregisteredApp?.appId
-				},
-				url: this._intentResolverOptions?.url,
-				frame: false,
-				autoShow: true,
-				alwaysOnTop: true
-			};
-
-			const win = await fin.Window.create(winOption);
-			const webWindow = win.getWebWindow();
 			try {
-				// eslint-disable-next-line @typescript-eslint/dot-notation, @typescript-eslint/no-explicit-any
-				const selectedAppId = await (webWindow as unknown as any)["getIntentSelection"]();
-				return selectedAppId as {
-					appId: string;
-					instanceId?: string;
-					intent: AppIntent;
-				};
+				const position = await getCenterPosition(clientIdentity, {
+					height: this._intentResolverOptions?.height ?? DEFAULT_INTENT_RESOLVER_HEIGHT,
+					width: this._intentResolverOptions?.width ?? DEFAULT_INTENT_RESOLVER_WIDTH
+				});
+
+				// TODO: Remove the following when the runtime is updated.
+				// The popped up window currently does not support interop if running under localhost.
+				// this is being fixed in a future runtime and in the interim we swap localhost for 127.0.0.1.
+				// other addresses work fine.
+				const intentPickerUrl = this._intentResolverOptions?.url.replace("://localhost", "://127.0.0.1");
+
+				const intentPickerResponse: OpenFin.PopupResult<IntentPickerResponse> = await fin.me.showPopupWindow({
+					additionalOptions: {
+						customData: {
+							title: this._intentResolverOptions?.title,
+							apps: launchOptions.apps,
+							intent: launchOptions.intent,
+							intents: launchOptions.intents,
+							unregisteredAppId: this._unregisteredApp?.appId
+						}
+					},
+					initialOptions: {
+						fdc3InteropApi: this._intentResolverOptions?.fdc3InteropApi,
+						defaultWidth: this._intentResolverOptions?.width,
+						defaultHeight: this._intentResolverOptions?.height,
+						showTaskbarIcon: false
+					},
+					url: intentPickerUrl,
+					resultDispatchBehavior: "close",
+					blurBehavior: "modal",
+					height: this._intentResolverOptions?.height,
+					width: this._intentResolverOptions?.width,
+					name: "intent-picker",
+					x: position?.x,
+					y: position?.y
+				});
+				if (isEmpty(intentPickerResponse.data)) {
```

Comment:
Is there a possibility that intentPickerResponse.data could legitimately be undefined ? You could check for `intentPickerResponse.result === "clicked"` instead